sources/api/migration/migrations/v1.6.0/vmware-host-containers/src/main.rs

```diff
@@ -0,0 +1,111 @@
+                        let prev_source = format!("{}:{}", ADMIN_CONTAINER_IMAGE_REPOSITORY, ver);
+                        if *source == prev_source {
+                            *source = format!(
+                                "{}:{}",
+                                ADMIN_CONTAINER_IMAGE_REPOSITORY, TARGET_ADMIN_CONTAINER_VERSION
+                            );
+                            println!(
+                                "Changed value of '{}' from '{}' to '{}' on upgrade",
+                                ADMIN_CONTAINER_SOURCE_SETTING_NAME, prev_source, source
+                            );
+                            break;
+                        }
+                    }
+                }
+                _ => {
+                    println!(
+                        "'{}' is set to non-string value '{}'",
+                        ADMIN_CONTAINER_SOURCE_SETTING_NAME, data
+                    );
+                }
+            }
+        } else {
+            println!(
+                "Found no '{}' to change on upgrade",
+                ADMIN_CONTAINER_SOURCE_SETTING_NAME
+            );
+        }
+
+        // For control container
+        if let Some(data) = input.data.get_mut(CONTROL_CONTAINER_SOURCE_SETTING_NAME) {
+            match data {
+                serde_json::Value::String(source) => {
+                    for ver in PREVIOUS_CONTROL_CONTAINER_VERSIONS {
+                        let prev_source = format!("{}:{}", CONTROL_CONTAINER_IMAGE_REPOSITORY, ver);
+                        if *source == prev_source {
+                            *source = format!(
+                                "{}:{}",
+                                CONTROL_CONTAINER_IMAGE_REPOSITORY,
+                                TARGET_CONTROL_CONTAINER_VERSION
+                            );
+                            println!(
+                                "Changed value of '{}' from '{}' to '{}' on upgrade",
+                                CONTROL_CONTAINER_SOURCE_SETTING_NAME, prev_source, source
+                            );
+                            break;
+                        }
+                    }
+                }
+                _ => {
+                    println!(
+                        "'{}' is set to non-string value '{}'",
+                        CONTROL_CONTAINER_SOURCE_SETTING_NAME, data
+                    );
+                }
+            }
+        } else {
+            println!(
+                "Found no '{}' to change on upgrade",
+                CONTROL_CONTAINER_SOURCE_SETTING_NAME
+            );
+        }
+
+        Ok(input)
+    }
+
+    fn backward(&mut self, input: MigrationData) -> Result<MigrationData> {
+        // It's unclear what version of the host-containers we should downgrade to since it could
+        // be any of the older host-container versions.
+        // We can just stay on the latest host-container version since there are no breaking changes.
```

Comment:
I want to open this up for discussion.
The alternative to this is to create separate migrations between all the older versions of bottlerocket with incremental host-container version bumps instead of just having a single migration that bumps all versions to latest. This is feasible with how we push updates and how migrations work. But I'm not sure what compelling reasons we may have for doing that.

Reply:
One reason might be for the 'enter-admin-container' script introduced in control-container v0.5.4. Since the script depends on `apicliet exec` which is a feature introduced in Bottlerocket version 1.4.0. The script wouldn't work on hosts older than v1.4.0.

Reply:
My vote: I think the bump from the existing version to the latest is fine.  On downgrade, I would expect my host containers to go back to the version they were before the upgrade.  If there is a specific reason for someone to use one of the incremental versions, that version could be supplied via setting.

Reply:
>  On downgrade, I would expect my host containers to go back to the version they were before the upgrade.
Unfortunately we can't know what the previously set host-container source is once we've migrated since it could be any of the older versions we migrated from. The only way to preserve that information is to have a chain of migrations, i.e. a migration between each Bottlerocket host version where we bumped the default host-container versions.

Reply:
> have a chain of migrations

This is probably the right thing to do.

Reply:
Hm. I have mixed feelings about this. I think in the instance that folks are downgrading and want a specific version, they should set it via user-data. The version of the container they had before would be dependent on the version when the host was initially set up, not necessarily the version they are rolling back to.